lib/baseProducer.js

```diff
@@ -134,7 +134,7 @@ BaseProducer.prototype.buildPayloads = function (payloads, topicMetadata) {
       if (message instanceof KeyedMessage) {
         return message;
       }
-      return new Message(0, 0, '', message);
+      return new Message(0, 0, p.key || '', message);
```

Comment:
hmm looks like that may not make it down the line checkout https://github.com/SOHU-Co/kafka-node/blob/master/lib/protocol/protocol.js#L412